resources/instance_term_protection.rb

```diff
@@ -0,0 +1,37 @@
+property :aws_access_key, String
+property :aws_secret_access_key, String
+property :aws_session_token, String
+property :aws_assume_role_arn, String
+property :aws_role_session_name, String
+property :region, String, default: lazy { fallback_region }
+
+include AwsCookbook::Ec2 # needed for aws_region helper
+
+action :enable do
+  unless term_protection_enabled?
+    converge_by('enable termination protection for this instance') do
+      ec2.modify_instance_attribute(instance_id: node['ec2']['instance_id'], disable_api_termination: { value: true })
```

Comment:
It would be nice to have an attribute for instance_id so that this could be done potentially from another instance. That would have a lazy eval for node['ec2']['instance_id'] so out of the box it would work the same